### ISSUE TITLE
Performance degradation occurs when global pruning with the random method

### ISSUE THREAD

**Issue by doriskim** [VainF/Torch-Pruning#498]
Since version 1.4.3, the RandomImportance() code has changed, which has caused a performance degradation in the global pruning of the random method. 
(The previous code showed over 95% accuracy, but now it shows about 50% accuracy.) 

Is this intentional?

### Before v1.4.3
```
class RandomImportance(Importance):
    @torch.no_grad()
    def __call__(self, group, **kwargs):
        _, idxs = group[0]
        return torch.rand(len(idxs))
```

### After v1.4.3

```
class RandomImportance(Importance):
    @torch.no_grad()
    def __call__(self, group, **kwargs):
        _, idxs = group[0]
        score = list(range(len(idxs)))
        random.shuffle(score)
        return torch.tensor(score, dtype=torch.float32)`
```

Before the change, a real random number was generated in the range of [0, 1) for each channel, but after the change, an integer random number is generated in the range of the number of channels in the group for each channel.

As a result, when performing global pruning in a random manner, groups with a small number of channels have a small importance value and become the main pruning target with low importance,
and channels with a high importance value in groups with a large number of channels are hardly pruned.

In this way, since each group has a different range of integer values, **an unfair comparison occurs in global pruning with the random method.**

**Comment by VainF** [comment]
I see... I think you are right! Will fix this in the next version.
